### ISSUE TITLE
Internal error: com.intellij.openapi.application.ReadAction$CannotReadException

### ISSUE THREAD

**Issue by angelozerr** [redhat-developer/intellij-quarkus#1032]
@ia3andy reports me the following error:

```
Internal error: com.intellij.openapi.application.ReadAction$CannotReadException

java.util.concurrent.CompletionException: com.intellij.openapi.application.ReadAction$CannotReadException
    at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:332)
    at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:347)
    at java.base/java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:708)
    at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
    at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2162)
    at com.redhat.devtools.intellij.lsp4ij.IndexAwareLanguageClient.runTask(IndexAwareLanguageClient.java:91)
    at com.redhat.devtools.intellij.lsp4ij.IndexAwareLanguageClient$1.run(IndexAwareLanguageClient.java:60)
    at com.intellij.openapi.progress.impl.CoreProgressManager.startTask(CoreProgressManager.java:429)
    at com.intellij.openapi.progress.impl.ProgressManagerImpl.startTask(ProgressManagerImpl.java:114)
    at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcessWithProgressInCurrentThread$11(CoreProgressManager.java:557)
    at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:186)
    at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$13(CoreProgressManager.java:604)
    at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:679)
    at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:635)
    at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:603)
    at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)
    at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:173)
    at com.intellij.openapi.progress.impl.CoreProgressManager.runProcessWithProgressInCurrentThread(CoreProgressManager.java:557)
    at com.intellij.openapi.progress.impl.CoreProgressManager.run(CoreProgressManager.java:363)
    at com.redhat.devtools.intellij.lsp4ij.IndexAwareLanguageClient.lambda$runAsBackground$0(IndexAwareLanguageClient.java:57)
    at com.redhat.devtools.intellij.lsp4ij.IndexAwareLanguageClient.lambda$runAsBackground$1(IndexAwareLanguageClient.java:71)
    at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1804)
    at java.base/java.util.concurrent.CompletableFuture$AsyncRun.exec(CompletableFuture.java:1796)
    at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
    at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
    at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
    at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
    at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
Caused by: com.intellij.openapi.application.ReadAction$CannotReadException
    at org.asciidoc.intellij.threading.AsciiDocProcessUtil.lambda$runInReadActionWithWriteActionPriority$0(AsciiDocProcessUtil.java:34)
    at org.asciidoc.intellij.threading.AsciiDocProcessUtil.retryable(AsciiDocProcessUtil.java:70)
    at org.asciidoc.intellij.threading.AsciiDocProcessUtil.runInReadActionWithWriteActionPriority(AsciiDocProcessUtil.java:30)
    at org.asciidoc.intellij.findUsages.AsciiDocJavaReferencesSearch.processQuery(AsciiDocJavaReferencesSearch.java:41)
    at org.asciidoc.intellij.findUsages.AsciiDocJavaReferencesSearch.processQuery(AsciiDocJavaReferencesSearch.java:30)
    at com.intellij.openapi.application.QueryExecutorBase.execute(QueryExecutorBase.java:76)
    at com.intellij.util.ExecutorsQuery.processResults(ExecutorsQuery.java:30)
    at com.intellij.util.AbstractQuery.doProcessResults(AbstractQuery.java:88)
    at com.intellij.util.AbstractQuery.delegateProcessResults(AbstractQuery.java:105)
    at com.intellij.util.MergeQuery.processResults(MergeQuery.java:22)
    at com.intellij.util.AbstractQuery.doProcessResults(AbstractQuery.java:88)
    at com.intellij.util.AbstractQuery.delegateProcessResults(AbstractQuery.java:105)
    at com.intellij.util.UniqueResultsQuery.processResults(UniqueResultsQuery.java:37)
    at com.intellij.util.AbstractQuery.doProcessResults(AbstractQuery.java:88)
    at com.intellij.util.AbstractQuery.delegateProcessResults(AbstractQuery.java:105)
    at com.intellij.util.MergeQuery.processResults(MergeQuery.java:22)
    at com.intellij.util.AbstractQuery.doProcessResults(AbstractQuery.java:88)
    at com.intellij.util.AbstractQuery.delegateProcessResults(AbstractQuery.java:105)
    at com.intellij.util.MergeQuery.processResults(MergeQuery.java:22)
    at com.intellij.util.AbstractQuery.doProcessResults(AbstractQuery.java:88)
    at com.intellij.util.AbstractQuery.delegateProcessResults(AbstractQuery.java:105)
    at com.intellij.util.MergeQuery.processResults(MergeQuery.java:22)
    at com.intellij.util.AbstractQuery.doProcessResults(AbstractQuery.java:88)
    at com.intellij.util.AbstractQuery.delegateProcessResults(AbstractQuery.java:105)
    at com.intellij.util.MergeQuery.processResults(MergeQuery.java:22)
    at com.intellij.util.AbstractQuery.doProcessResults(AbstractQuery.java:88)
    at com.intellij.util.AbstractQuery.delegateProcessResults(AbstractQuery.java:105)
    at com.intellij.util.MergeQuery.processResults(MergeQuery.java:22)
    at com.intellij.util.AbstractQuery.doProcessResults(AbstractQuery.java:88)
    at com.intellij.util.AbstractQuery.forEach(AbstractQuery.java:80)
    at com.intellij.util.AbstractQuery.findAll(AbstractQuery.java:25)
    at com.intellij.util.Query.iterator(Query.java:127)
    at com.intellij.util.AbstractQuery.iterator(AbstractQuery.java:36)
    at java.base/java.lang.Iterable.forEach(Iterable.java:74)
    at com.redhat.devtools.intellij.qute.psi.internal.template.datamodel.DataModelProviderRegistry.scanJavaClasses(DataModelProviderRegistry.java:175)
    at com.redhat.devtools.intellij.qute.psi.internal.template.datamodel.DataModelProviderRegistry.collectDataModel(DataModelProviderRegistry.java:146)
    at com.redhat.devtools.intellij.qute.psi.internal.template.datamodel.DataModelProviderRegistry.getDataModelProject(DataModelProviderRegistry.java:123)
    at com.redhat.devtools.intellij.qute.psi.internal.template.QuarkusIntegrationForQute.getDataModelProject(QuarkusIntegrationForQute.java:64)
    at com.redhat.devtools.intellij.qute.psi.QuteSupportForTemplate.getDataModelProject(QuteSupportForTemplate.java:118)
    at com.redhat.devtools.intellij.qute.lsp.QuteLanguageClient.lambda$getDataModelProject$4(QuteLanguageClient.java:140)
    at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:898)
    at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:68)
    at com.redhat.devtools.intellij.qute.lsp.QuteLanguageClient.lambda$getDataModelProject$5(QuteLanguageClient.java:140)
    at com.redhat.devtools.intellij.lsp4ij.IndexAwareLanguageClient.lambda$runTask$2(IndexAwareLanguageClient.java:83)
    at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:885)
    at com.redhat.devtools.intellij.lsp4ij.IndexAwareLanguageClient.runTask(IndexAwareLanguageClient.java:83)
    ... 22 more

```

That's strange, I though @fbricon takes care of this usecase.

**Comment by angelozerr** [comment]
I notice that there are 2 runReadActio, is it the problem?

at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:898)
    at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:68)
    at com.redhat.devtools.intellij.qute.lsp.QuteLanguageClient.lambda$getDataModelProject$5(QuteLanguageClient.java:140)
    at com.redhat.devtools.intellij.lsp4ij.IndexAwareLanguageClient.lambda$runTask$2(IndexAwareLanguageClient.java:83)
    at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:885)

**Comment by fbricon** [comment]
should be fixed by #1072
